### PR TITLE
[stable/ndm]: add node selector and toleration to ndm exporter

### LINKED ISSUE #301: node selection not available for openebs-ndm:ndmExporter
Hi,

I am trying to install openebs chart on my cluster's specific node.
The options for tolerations and affinity are not available for openebs-ndm:ndmExporter.

Every other component which I am using has this option available.

Would be great if this can be accommodated in the next release.

Thanks,

### PR DESCRIPTION
**Pull request by akhilerm**
- add toleration and node selector for NDM exporter pods
- update chart version
- update values.yaml
- update README

Fixes : openebs/charts#301

Signed-off-by: Akhil Mohan <akhil.mohan@datacore.com>

### CODE REVIEW

**Review by Ab-hishek — DISMISSED** [review]

File: deploy/helm/charts/templates/cluster-exporter.yaml [hunk 1]
```diff
@@ -45,4 +45,16 @@ spec:
             - name: METRICS_LISTEN_PORT
               value: :{{ .Values.ndmExporter.clusterExporter.metricsPort }}
             {{- end }}
-{{- end }}
\ No newline at end of file
+{{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 6 }}
```
Comment (z0marlin) [inline deploy/helm/charts/templates/cluster-exporter.yaml]: why do we need to add a new line here but not for the rest of the indents?

**Review by z0marlin — DISMISSED** [review]

File: deploy/helm/charts/templates/cluster-exporter.yaml (hunk 1 above)
Comment (akhilerm) [reply deploy/helm/charts/templates/cluster-exporter.yaml]: We add the new line everywhere. Or did I misunderstood the question.

Comment (z0marlin) [reply deploy/helm/charts/templates/cluster-exporter.yaml]: I mean the new line through `nindent`. For others we are using `indent`

Comment (xunholy) [reply deploy/helm/charts/templates/cluster-exporter.yaml]: I'd rather this was nindent 8 to keep it consistent with existing list fields.

File: deploy/helm/charts/templates/cluster-exporter.yaml [hunk 2]
```diff
@@ -45,4 +45,16 @@ spec:
...
+{{- end }}
+{{- if .Values.ndmExporter.clusterExporter.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.ndmExporter.clusterExporter.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.ndmExporter.clusterExporter.tolerations }}
+      tolerations:
+{{ toYaml .Values.ndmExporter.clusterExporter.tolerations | indent 8 }}
```
Comment (xunholy) [inline deploy/helm/charts/templates/cluster-exporter.yaml:58]: For consistency should these be indented similar to LOC 50 in your proposal? Applicable in all other places, either way as long as we keep it consistent.

File: deploy/helm/charts/templates/cluster-exporter.yaml (hunk 1 above)
Comment (akhilerm) [reply deploy/helm/charts/templates/cluster-exporter.yaml]: updated

File: deploy/helm/charts/templates/cluster-exporter.yaml (hunk 2 above)
Comment (akhilerm) [reply deploy/helm/charts/templates/cluster-exporter.yaml:58]: updated

**Comment by akhilerm** [comment]
Updated the indent in image pull secrets. @sonasingh46 Do you remember why we made image pull secrets different from other keys like tolerations and nodeselector. When I tested, I see that both of the methods are working.

**Review by z0marlin — APPROVED** [review]
LGTM

**Comment by akhilerm** [comment]
@xUnholy and @Ab-hishek can you take a look at the PR?

**Review by Ab-hishek — APPROVED** [review]
LGTM

**Comment by pawanpraka1** [comment]
@akhilerm can we merge this one? RC1 branching is already done.

**Pull request merged**